src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -616,7 +616,6 @@ private void CheckStake(Block block, ChainedHeader chainTip)
         {
             this.logger.LogTrace("({0}.{1}:{2},{3}:'{4}',{5}:{6},{7}:{8})", nameof(utxoStakeDescriptions), nameof(utxoStakeDescriptions.Count), utxoStakeDescriptions.Count, nameof(chainTip), chainTip, nameof(searchInterval), searchInterval, nameof(fees), fees);
 
-            int nonEmptyUtxos = utxoStakeDescriptions.Count;
```

Comment:
Here I have removed the use of Non Empty UTXOs and instead use only the count of coins that have been picked up for staking, let me know if you think that is not correct (and why :) !)

Reply:
I don't think this is correct. The name is slightly wrong because of our previous PR, so this value wouldn't contain all nonempty UTXOs, but it would contain only those that are bigger than the minimum, but that is slightly irrelevant. What is relevant that we want to include UTXOs that we currently cannot use because of their age or maturity. Therefore we should not use the number of coins that can be used at this point for staking